### DISCUSSION TITLE
Library: Live photo / video preview loading delay (1-2 seconds)

### DISCUSSION THREAD

**Discussion by vandad51** [photoprism/photoprism#4065] [Q&A]
**Describe what problem this solves and why this would be valuable to many users**
-
While on my desktop, and having the PhotoPrism log open in another screen. As I move my mouse around the screen, the logs starts piling up with Transcoding requests for all the pictures and videos that do not have any rendered video clips.

What's happening is as I'm scrolling through my library, my CPU is now saturated with requests for transcoding that I do not want happening. It maxes out at 100% of all cores assigned to PhotoPrism (using Unraid)

**Describe the solution you'd like**
-
I would like to see a setting for when browsing thumbnails, that I can choose to either transcode everything my mouse touches, or wait for a 1-2 second hover before beginning the transcode request.

**Alternate solution**
Maybe these previews could be automatically generated upon indexing?

**Comment by lastzero** [comment]
**If necessary, videos are transcoded on demand. This can cause unacceptable delays when large video files are played for the first time.**

In this case, you can [run the following command in a terminal](https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface) to pre-transcode all video files as needed:

```bash
docker compose exec photoprism photoprism convert
```

For further details, see: https://docs.photoprism.app/user-guide/organize/video/#transcoding

**Reply by vandad51** [reply]
@lastzero Thanks for the reply, I did forget about transcoding upon import.

My issue is not that there's delays when loading a video for the first time on Photoprism, my Issue is that when I have live photos (iOS) all over my visible screen that are not transcoded yet, as I move my mouse around or scroll to find a photo I want to view, it starts transcoding every single video and live photo clip my mouse touched - which cripples photoprism until the que of what could be up to 100 video, is complete which allows Photoprism to come back to life.

I have 6 of 8 threads dedicated to Photoprism docker on Unraid and they are at 100% while this is happening, even navigating to other parts of photoprism is shockingly slow.

This is where my suggestion of a 1-2 second delay option would be helpful to not pile up the transcoder que and cripple Photoprism's functionality. If the preview video is transcoded, play it, otherwise, wait 1-2 seconds for confirmation that transcoding is requested by intentional user input.

**Reply by lastzero** [reply]
First you say you didn't forget about transcoding and then you say the videos are not transcoded yet? Wouldn't it be best to pre-transcode your videos and live photos to avoid any delay and reduce CPU usage while you browse your pictures?
